### PR TITLE
create rasa credentials if not given

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fixes the "Input is not to a terminal (fd=4)." error if no credentials are given
- automatically inserts `rasa` channel credentials if none are present

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/cli/x.py [hunk 1]
```diff
@@ -92,16 +93,14 @@ def is_metrics_collection_enabled(args: argparse.Namespace) -> bool:
 
 
 def _core_service(args: argparse.Namespace, endpoints: "AvailableEndpoints" = None):
```
Comment (ricwo) [inline rasa/cli/x.py]:
```suggestion
 def _rasa_service(args: argparse.Namespace, endpoints: "AvailableEndpoints" = None):
```

File: rasa/cli/x.py [hunk 2]
```diff
@@ -140,8 +163,7 @@ def start_core_for_local_platform(args: argparse.Namespace, rasa_x_token: Text):
     vars(args).update(
         dict(
             nlu_model=None,
-            channel="rasa",
-            credentials="credentials.yml",
+            credentials=DEFAULT_CREDENTIALS_PATH,
```
Comment (ricwo) [inline rasa/cli/x.py]: this shouldn't necessarily take the default should it?
```suggestion
            credentials=credentials_path,
```

File: tests/cli/test_rasa_x.py [hunk 3]
```diff
@@ -15,3 +19,33 @@ def test_x_help(run):
 
     for i, line in enumerate(lines):
         assert output.outlines[i] == line
+
+
+def test_prepare_credentials_for_rasa_x_if_rasa_channel_not_given(tmpdir_factory):
+    directory = tmpdir_factory.mktemp("directory")
+    credentials_path = str(directory / "credentials.yml")
```
Comment (ricwo) [inline tests/cli/test_rasa_x.py]: nice 💯

File: rasa/cli/x.py [hunk 4]
```diff
@@ -91,17 +92,15 @@ def is_metrics_collection_enabled(args: argparse.Namespace) -> bool:
     return allow_metrics
 
 
-def _core_service(args: argparse.Namespace, endpoints: "AvailableEndpoints" = None):
-    """Starts the Rasa Core application."""
+ def _rasa_service(args: argparse.Namespace, endpoints: "AvailableEndpoints" = None):
```
Comment (ricwo) [inline rasa/cli/x.py]:
```suggestion
def _rasa_service(args: argparse.Namespace, endpoints: "AvailableEndpoints" = None):
```

File: rasa/cli/x.py (hunk 2 above)
Comment (wochinge) [reply rasa/cli/x.py]: I think we can just drop it

Comment (tmbo) [reply rasa/cli/x.py]: yes

**Pull request merged**